code/game/machinery/hologram.dm

```diff
@@ -704,3 +704,80 @@ For the other part of the code, check silicon say.dm. Particularly robot talk.*/
 
 #undef HOLOPAD_PASSIVE_POWER_USAGE
 #undef HOLOGRAM_POWER_USAGE
+
+/obj/machinery/holopad/proc/create_admin_hologram(client/C)
+	RETURN_TYPE(/mob/living/simple_animal/admin_holopad)
+	var/icon/final = icon()
+	var/mob/living/carbon/human/dummy/D = new(locate(1,1,1)) //spawn on 1,1,1 so we don't have runtimes when items are deleted
+	var/new_name = null //Name for the hologram. Can be defaulted to your char name otherwise.
+	if(alert("Do you want to use your character slot ([C.prefs.real_name]) or choose a new name and have a random appearance?",src,"Yes","No") == "Yes")
+		C.prefs.copy_to(D)
+		new_name = C.prefs.real_name
+	else
+		randomize_human(D)
+		new_name = input(C, "Select a name for your communications hologram (leave blank to just be called hologram)", "Robust hologram creator") as text
+	C.cmd_admin_dress(D)
+	for(var/dir in list(2, 1, 4, 8)) //A little bit hacky, but does the trick. BYOND's icon format starts with a south facing dir, while cardinals starts with NORTH.
+		CHECK_TICK
+		D.setDir(dir) //Set the dir of the mannequin to face the next direction
+		COMPILE_OVERLAYS(D) //Prepare it for an image capture
+		var/icon/I = icon(getFlatIcon(D), frame = 1) //And finally clone the appearance of our new dummy character.
+		final.Insert(I, "hologram", frame=1, dir=dir) //Then, we add this new icon_state and direction to the icon we're generating. This is then cleanly applied to the dummy mob to give it its appearance.
+	qdel(D)
```

Comment:
I hate this making a human and then immediately deleting them, but I guess thats how its done anyway